### PR TITLE
DOC: Updates to documentation

### LINKED ISSUE #48674: BUG: Wrong index constructor if new DF/Series is created from DF/Series
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
from pandas import DataFrame as pDF, Series as pSe

# INIT
# ────────────────────────────────────────────────────────────
# DATA
x = pDF({'A':[1,2,3,4],'B':[5,6,7,8]}, index=[1,0,1,0])

# NEW VAR TO BE ADDED IN SEVERAL WAYS
N = [1,2,3,4]


# (1) Works as expected → New variable from a List
# ────────────────────────────────────────────────────────────
x['N'] = N

# (1) RESULT → OK
	A	B	N
1	1	5	1
0	2	6	2
1	3	7	3
0	4	8	4


# (2) Works as expected → New variable from pSe/PDF
# ────────────────────────────────────────────────────────────
x['N'] = pDF(N, index=x.index)
# OR 
x['N'] = pSe(N, index=x.index)
# Produces the same correct result

# (2) RESULT → OK
	A	B	N
1	1	5	1
0	2	6	2
1	3	7	3
0	4	8	4

# (3) DOESN'T Work as expected → New variable from ► INNER pSe/PDF ◄
# ────────────────────────────────────────────────────────────
x['N'] = pDF(pSe(N), index=x.index)
# OR 
x['N'] = pSe(pSe(N), index=x.index)
# Produces WRONG Result as index mismatch

# (3) RESULT → WRONG → INDEX MISMATCH
	A	B	N
1	1	5	2
0	2	6	1
1	3	7	2
0	4	8	1

# (4) WRONG behavior can be overcome by using set_index as follows:
x['N'] = pDF(pSe(N)).set_index(x.index)
# Which produces correct result
```

### Issue Description

Working with variables, and creating new variables is a crucial component of pandas. 
Thus I assume that the construct of the new variable should be consistent regardless of using index= or set_index() method.

The example above shows that issue is inside the constructor when an inner dataframe or series is used and a non-unique index is present in the targeted dataframe → in such a situation using index= param during the constructor doesn't work as expected and it is inconsistent with the rest of the approachs.

### Expected Behavior

Consistent behavior of constructor for index= and set_index regardless of input object to DataFrame/Serie.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : e8093ba372f9adfe79439d90fe74b0b5b6dea9d6
python           : 3.9.7.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19043
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 13, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252

pandas           : 1.4.3
numpy            : 1.20.3
pytz             : 2021.3
dateutil         : 2.8.2
setuptools       : 58.0.4
pip              : 21.2.4
Cython           : 0.29.24
pytest           : 6.2.4
hypothesis       : None
sphinx           : 4.2.0
blosc            : None
feather          : None
xlsxwriter       : 3.0.1
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : 1.3.2
brotli           : 
fastparquet      : None
fsspec           : 2021.10.1
gcsfs            : None
markupsafe       : 1.1.1
matplotlib       : 3.5.2
numba            : 0.54.1
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 7.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.0
snappy           : None
sqlalchemy       : 1.4.22
tables           : 3.6.1
tabulate         : None
xarray           : None
xlrd             : 2.0.1
xlwt             : 1.3.0
zstandard        : None
</details>

### PR DESCRIPTION
**Pull request by minat-hub**
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `pandas/core/frame.py` file.
- [ ] closes #48674 

index constructor updates to the frame.py file to indicate index alignment.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -121,6 +121,7 @@ doc/build/html/index.html
 # Windows specific leftover:
 doc/tmp.sv
 env/
+pandas-env
```
Comment (mroeschke) [inline .gitignore:124]: Could you undo this change?

File: pandas/core/frame.py [hunk 2]
```diff
@@ -592,6 +593,25 @@ class DataFrame(NDFrame, OpsMixin):
     0  0  0
     1  0  3
     2  2  3
+
+    Constructing DataFrame from Series/DataFrame:
+
+    >>> ser = pd.Series([1, 2, 3], index=["a", "b", "c"])
+    >>> df = pd.DataFrame(data=ser, index=["a", "c"])
+    >>> print(df)
```
Comment (mroeschke) [inline pandas/core/frame.py:601]:
```suggestion
    >>> df
```

File: pandas/core/frame.py [hunk 3]
```diff
@@ -592,6 +593,25 @@ class DataFrame(NDFrame, OpsMixin):
...
+
+       0
+    a  1
+    c  3
+
+    >>> df1 = pd.DataFrame([1, 2, 3], index=["a", "b", "c"], columns=["x"])
+    >>> df2 = pd.DataFrame(data=df1, index=["a", "c"])
+    >>> print(df2)
```
Comment (mroeschke) [inline pandas/core/frame.py:609]:
```suggestion
    >>> df2
```

**Comment by mroeschke** [comment]
Thanks for the pull request. Could you please review the CI failures?

**Comment by minat-hub** [comment]
> Thanks for the pull request. Could you please review the CI failures?

Okay, Thank you @mroeschke. I will review and reopen the pull request after making the requested changes

**Comment by minat-hub** [comment]
@mroeschke All ci checks are successful now and review has been implemented. Thank you for the feedback.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Great, thanks @minat-hub
